protocol/messenger.go

```diff
@@ -258,18 +258,27 @@ func (m *Messenger) ResolvePrimaryName(mentionID string) (string, error) {
 // EnvelopeSent triggered when envelope delivered at least to 1 peer.
 func (interceptor EnvelopeEventsInterceptor) EnvelopeSent(identifiers [][]byte) {
 	if interceptor.Messenger != nil {
-		var ids []string
+		signalIDs := make([][]byte, 0, len(identifiers))
 		for _, identifierBytes := range identifiers {
-			ids = append(ids, types.EncodeHex(identifierBytes))
-		}
+			messageID := types.EncodeHex(identifierBytes)
+			err := interceptor.Messenger.processSentMessage(messageID)
+			if err != nil {
+				interceptor.Messenger.logger.Info("messenger failed to process sent messages", zap.Error(err))
+			}
 
-		err := interceptor.Messenger.processSentMessages(ids)
-		if err != nil {
-			interceptor.Messenger.logger.Info("messenger failed to process sent messages", zap.Error(err))
+			message, err := interceptor.Messenger.MessageByID(messageID)
+			if err != nil {
+				interceptor.Messenger.logger.Error("failed to query message outgoing status", zap.Error(err))
+			} else {
+				if message.OutgoingStatus == common.OutgoingStatusDelivered {
+					// We don't want to send the signal if the message was already marked as delivered
+					continue
+				} else {
+					signalIDs = append(signalIDs, identifierBytes)
+				}
+			}
```

Comment:
No need for those `else`s, they only make it harder to read
```suggestion
			message, err := interceptor.Messenger.MessageByID(messageID)
			if err != nil {
				interceptor.Messenger.logger.Error("failed to query message outgoing status", zap.Error(err))
				continue
			}
		    if message.OutgoingStatus == common.OutgoingStatusDelivered {
			    // We don't want to send the signal if the message was already marked as delivered
			    continue
		    }
			signalIDs = append(signalIDs, identifierBytes)
```